cmd/podman/shared/create.go

```diff
@@ -83,7 +83,7 @@ func CreateContainer(ctx context.Context, c *GenericCLIResults, runtime *libpod.
 		} else {
 			return nil, nil, errors.Errorf("error, no input arguments were provided")
 		}
-		newImage, err := runtime.ImageRuntime().New(ctx, name, rtc.SignaturePolicyPath, GetAuthFile(""), writer, nil, image.SigningOptions{}, false, nil)
+		newImage, err := runtime.ImageRuntime().New(ctx, name, rtc.SignaturePolicyPath, GetAuthFile(""), writer, nil, image.SigningOptions{}, nil, c.String("pull"))
```

Comment:
I can’t see any code that validates the CLI argument to be one of the three valid values.

Reply:
you mean check if this c.Stirng("pull") is always missing or never? https://github.com/containers/libpod/pull/3617/files#diff-3303ff4a5ad91328550c7bee8df0be69R157

Reply:
:ok I know I'll try to add this

Reply:
Yes. Nothing in that code rejects `"this-is-completely-invalid"` AFAICS; it silently treats it as `PullImageMissing`.

We should clearly reject unexpected values, both to protect users against typos and to make it possible to add more values in the future.

(I’d prefer to have a typed enum for the parameters everywhere, and handle the validation at the CLI level, or somewhere close, using a helper function that converts from untyped/unverified strings into the typed enums.

Using a typed enum would also allow using integers for the underlying representation, which is a tiny bit more efficient — but this micro-optimization is by no means the decisive factor; reliability and clarity is much more important.)